### PR TITLE
Upgrade wizard with checkboxss

### PR DESCRIPTION
**Pull request by akucharska**
**Description**

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

### CODE REVIEW

**Review by Akatuoro — COMMENTED** [review]
There could be a few index discrepancies:

The link uses `resource.spec.modules[index]` while the Checkbox setter uses `resource.spec.modules.findIndex(module => module.name === name)`

If we assume that each module has a unique name, I think it would be good to fix the module index and the module object for consistency. E.g.:
```js
const moduleIndex = resource?.spec?.modules?
    resource.spec.modules.findIndex(module => module.name === name) :
    index;
const module = resource?.spec?.modules?.[moduleIndex];
```


If we assume that `moduleTemplate` has a unique name, then we could also fixate that. But for the `channelTests`, it seems that there could be multiple module templates with the same name, but different channels.

If so, then it'd be good to fixate the specific module template that matches the currently selected channel, e.g.:
```js
const channelModuleTemplate = parsedOptions?.moduleTemplates?.find(moduleTemplate => {
    if (moduleTemplate?.metadata?.labels['operator.kyma-project.io/module-name'] === name && moduleTemplate?.spec?.channel === channel) return moduleTemplate;
    return null;
});
```

The `isBeta` variable is currently based on any `moduleTemplate` with the module name, not considering the selected channel. From my understanding, this should be changed to consider both the name and channel.

File: src/components/Extensibility/components-form/Modules/Modules.js [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+          });
+
+        return channelTest.push({
+          text: moduleTemplate.spec.channel,
+          additionalText: '',
+        });
+      } else {
+        return '';
+      }
+    });
```
Comment (Akatuoro) [inline src/components/Extensibility/components-form/Modules/Modules.js:108]: Could be simplified to a filter and map
```suggestion
    const channelTest = parsedOptions?.moduleTemplates?.filter(moduleTemplate => moduleTemplate?.metadata?.labels['operator.kyma-project.io/module-name'] === name).map(moduleTemplate => {
        const additionalText = moduleTemplate?.spec?.descriptor?.component?.version?
            `(version: ${moduleTemplate?.spec?.descriptor?.component?.version})` :
            '';
        return channelTest.push({
          text: moduleTemplate.spec.channel,
          additionalText,
        });
    });
```

File: src/components/Extensibility/components-form/Modules/Modules.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        moduleTemplate?.metadata?.labels['operator.kyma-project.io/beta'] ===
+          'true',
+    );
+
+    return (
+      <>
+        <div className="flexbox fd-margin-bottom--sm">
+          <div className="fd-color--neutral">
+            {index === 0 ? `${sectionName}:` : ''}
+          </div>
```
Comment (Akatuoro) [inline src/components/Extensibility/components-form/Modules/Modules.js:139]: sectionName is already in the collapsible section part below

File: src/components/Extensibility/components-form/Modules/Modules.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+                      return module.name === name;
+                    })
+                  : index,
+              );
+            }}
+          >
+            {name}
+          </Checkbox>
+          <div>
+            <p className="fd-color--neutral">Module channel overwrite:</p>
```
Comment (Akatuoro) [inline src/components/Extensibility/components-form/Modules/Modules.js:161]: translation missing

File: src/components/Extensibility/components-form/Modules/Modules.js [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+            <ComboboxInput
+              compact
+              disabled={!isChecked}
+              placeholder={'Uses Default Kyma Channel'}
```
Comment (Akatuoro) [inline src/components/Extensibility/components-form/Modules/Modules.js:165]: translation missing

File: src/components/Extensibility/components-form/Modules/Modules.js [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+          </div>
+
+          {!parsedOptions?.displayDocs && link ? (
+            <Link
+              href={link}
+              target="_blank"
+              rel="noopener noreferrer"
+              className="fd-align-vertical-center"
+            >
+              DOCUMENTATION
```
Comment (Akatuoro) [inline src/components/Extensibility/components-form/Modules/Modules.js:204]: translation missing

**Review by mrCherry97 — DISMISSED** [review]

**Review by mrCherry97 — APPROVED** [review]

**Review by mrCherry97 — APPROVED** [review]

**Review by nataliasitko — APPROVED** [review]

**Pull request merged**